### DISCUSSION TITLE
Exchange federation question

### DISCUSSION THREAD

**Discussion by kotyara85** [rabbitmq/rabbitmq-server#3236] [Other]
Hello,
I have two separate clusters and want to federate them to fanout messages from one cluster to another.
I did setup exchanges, policies, upstreams, but when I send a message from cluster1 -> cluster2 I don't see this message in demo queue on cluster2

Any suggestions?

Thank you

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
I't s a bit difficult to suggest much since you haven't shared any specific steps that were taken.
If you seek community, leave alone the core team help, please help others help you and share as much relevant information as possible. We do not guess in this community: guessing is a very expensive approach to troubleshooting.

First I'd verify is that your exchanges are actually federated. This means they must be matched by a policy and the downstream side must **open connections** (links) to the upstream. If that does not happen, then the most likely reasons are

 * Policies were applied on the wrong side
 * The policy that matches the exchanges does not enable federation ([only one policy matches](https://rabbitmq.com/parameters.html#how-policies-work) an object at any given time)

Federation links will be marked as such with an appropriate subscript "client type" on the connection page.

Assuming that exchange federation links run and there is a message flow on the **upstream side**, the links should demonstrate
traffic flow and the federated (downstream) exchanges should have > 0 ingress message rates. From there, routing would work
exactly the same as it would without federation. Perhaps the only difference is that the `rabbitmq_federation` plugin will propagate bindings
for the federated exchanges when it can do so.

**Reply by kotyara85** [reply]
![Screen Shot 2021-07-26 at 4 17 57 PM](https://user-images.githubusercontent.com/10145395/127071174-12103a59-0ebb-4c35-8bf4-a1f69c0a2f30.png)
This's how my topology looks like. 
Links are up and I do see active connections between the clusters.

Exchange configuration on both sides
![Screen Shot 2021-07-26 at 4 19 55 PM](https://user-images.githubusercontent.com/10145395/127071296-329f0ad6-53d7-47cd-930d-c713209e71b6.png)

Policies
![Screen Shot 2021-07-26 at 4 19 07 PM](https://user-images.githubusercontent.com/10145395/127071243-b2d69eb3-1cc3-4434-9833-b8b2eb792b4e.png)

I can send messages directly from cluster 2 cluster if I click on demo@cluster2 queue for example, but that's it

Thank you

**Reply by michaelklishin** [reply]
According to the exchange page, it is not matched by any policies.

**Reply by michaelklishin** [reply]
Since the name does match the pattern, quite likely that you have other policies that match the same object with a higher (or equal) priority. Alternatively this policy may be applied on the wrong side: federation links are started **by the downstream** for every exchange or queue that is **matched by a policy** that enables federation (by including an upstream or upstream set name).

**Comment by michaelklishin** [comment]
Here's what an exchange matched by a policy looks like on the downstream, together with a link displayed when `rabbitmq_federation_management` is enabled:

<img width="589" alt="Screenshot 2021-07-27 at 02 34 24" src="https://user-images.githubusercontent.com/1090/127072439-147e8746-15e1-465a-a688-aadceb816f08.png">
<img width="595" alt="Screenshot 2021-07-27 at 02 34 42" src="https://user-images.githubusercontent.com/1090/127072444-f1224c45-2c03-40fc-836a-7154841ca406.png">
<img width="973" alt="Screenshot 2021-07-27 at 02 34 54" src="https://user-images.githubusercontent.com/1090/127072446-145e9eca-8a4f-4113-b9a0-67635f2d707e.png">

On the upstream side, you can see a federation link as an inbound connection and a grayed out federated exchange on the exchange list:

<img width="916" alt="Screenshot 2021-07-27 at 02 35 04" src="https://user-images.githubusercontent.com/1090/127072503-d701bac2-4db4-416c-916e-07f0ce47f65b.png">
<img width="831" alt="Screenshot 2021-07-27 at 02 35 13" src="https://user-images.githubusercontent.com/1090/127072504-a1743266-dfce-4351-b922-580d2677ef68.png">

**Comment by kotyara85** [comment]
Yeah, all of these are present in my setup.

![Screen Shot 2021-07-26 at 4 39 26 PM](https://user-images.githubusercontent.com/10145395/127072685-77732d98-7c07-4b16-bed1-cd77e4cd9c44.png)

![Screen Shot 2021-07-26 at 4 40 38 PM](https://user-images.githubusercontent.com/10145395/127072787-d3430e4b-d8cf-44fd-9e11-05ff03bed671.png)


![Screen Shot 2021-07-26 at 4 40 53 PM](https://user-images.githubusercontent.com/10145395/127072806-d115e49d-a539-47ff-a2cd-cece14cc2b8e.png)


Not sure what else could be wrong
Thanks

**Reply by kotyara85** [reply]
Okay, so I started from the scratch and it seems to be working now, thanks!
